### ISSUE TITLE
RSpec/VerifiedDoubles requires instance_double in cases where duck typing is employed

### ISSUE THREAD

**Issue by CJBridges** [nevir/rubocop-rspec#36]
Duck typing - double stubs a method, but could be one of many classes *with no shared base class*

```ruby
def Runner
  def initialize(worker)
    @worker = worker
  end

  def call
    worker.call
  end
end
```

A spec for this could look something like...
```ruby
describe 'Runner' do
  let(:return_val) { 'RetVal' }
  let(:my_worker) { double(:sample_worker, call: return_val) } # ack, verified double checker

  it 'returns the result of my_worker.call'
    expect(Runner.new(my_worker).call).to be(return_val)
  end
end
```

*Proposed fix*: ??? unknown here.

*Workaround*: Define something like this instead in the spec file:
```ruby
class Spec::ExampleWorker
  def call; end
end
```

Then change the reference to:
```ruby
instance_double('Spec::ExampleWorker')
```

**Comment by CJBridges** [comment]
P.S. These were real world problems encountered while trying to enable the verified double checker (which I heartily approve of!)

**Comment by andyw8** [comment]
I would be inclined to choose an arbitrary existing class that implemented the particular interface.

**Comment by CJBridges** [comment]
I like the view that tests can serve documentation of the interface of the class and not really tied to any particular caller (which may vanish completely separate from the class).

I tried a couple approaches including OpenStruct for some simple cases or creating a real instance that behaves like I wanted to... something like... `Object.new.tap { |o| def o.method; 'foo'; end }`  (this is tough since the body of the method doesn't evaluate in rspec context, harder to vary values)

Overall, I haven't found anything that resolves the issue any better than the workaround I noted in the description, unfortunately.  The joys of duck typing!

**Comment by backus** [comment]
@andyw8 similar to #35 it should be alright to close this, correct?

**Comment by andyw8** [comment]
Yes, resolved by #99.
